watchtower/blob/justice_kit.go

```diff
@@ -267,3 +270,208 @@ func (a *anchorJusticeKit) ToRemoteOutputSpendInfo() ([]byte, [][]byte, uint32,
 
 	return toRemoteScriptHash, witness, 1, nil
 }
+
+// taprootJusticeKit is an implementation of the JusticeKit interface which can
+// be used for backing up commitments of taproot channels.
+type taprootJusticeKit struct {
+	justiceKitPacketV1
+}
+
+// A compile-time check to ensure that taprootJusticeKit implements the
+// JusticeKit interface.
+var _ JusticeKit = (*taprootJusticeKit)(nil)
+
+// newTaprootJusticeKit constructs a new anchorJusticeKit.
+func newTaprootJusticeKit(sweepScript []byte,
+	breachInfo *lnwallet.BreachRetribution,
+	withToRemote bool) (*taprootJusticeKit, error) {
+
+	keyRing := breachInfo.KeyRing
+
+	tree, err := input.NewLocalCommitScriptTree(
+		breachInfo.RemoteDelay, keyRing.ToLocalKey,
+		keyRing.RevocationKey,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	packet := justiceKitPacketV1{
+		sweepAddress: sweepScript,
+		revocationPubKey: toBlobSchnorrPubKey(
+			keyRing.RevocationKey,
+		),
+		localDelayPubKey: toBlobSchnorrPubKey(keyRing.ToLocalKey),
+		delayScriptHash:  tree.SettleLeaf.TapHash(),
+	}
+
+	if withToRemote {
+		packet.commitToRemotePubKey = toBlobPubKey(keyRing.ToRemoteKey)
+	}
+
+	return &taprootJusticeKit{packet}, nil
+}
+
+// ToLocalOutputSpendInfo returns the info required to send the to-local
+// output. It returns the output pub key script and the witness required
+// to spend the output.
+//
+// NOTE: This is part of the JusticeKit interface.
+func (t *taprootJusticeKit) ToLocalOutputSpendInfo() ([]byte, [][]byte, error) {
+	revocationPubKey, err := schnorr.ParsePubKey(t.revocationPubKey[:])
+	if err != nil {
+		return nil, nil, err
+	}
+
+	localDelayedPubKey, err := schnorr.ParsePubKey(t.localDelayPubKey[:])
+	if err != nil {
+		return nil, nil, err
+	}
+
+	revokeScript, err := input.NewLocalCommitRevokeScript(
+		localDelayedPubKey, revocationPubKey,
+	)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	revokeLeaf := txscript.NewBaseTapLeaf(revokeScript)
+	revokeLeafHash := revokeLeaf.TapHash()
+	rootHash := tapBranchHash(revokeLeafHash[:], t.delayScriptHash[:])
+
+	outputKey := txscript.ComputeTaprootOutputKey(
+		&input.TaprootNUMSKey, rootHash[:],
+	)
+
+	scriptPk, err := input.PayToTaprootScript(outputKey)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var outputKeyYIsOdd bool
+	if outputKey.SerializeCompressed()[0] ==
```

Comment:
We have a check like this in so many places that we should make a helper function for it: 
```go
func IsOddPub(btcec.PublicKey) bool
```

Reply:
Gotcha, yeah we really need this in `btcec/v2` actually.